src/libstore/references.hh

```diff
@@ -27,6 +27,22 @@ public:
     void operator () (std::string_view data) override;
 };
 
+class PathRefScanSink : public RefScanSink
+{
+    std::map<std::string, StorePath> backMap;
+
+    PathRefScanSink(StringSet && hashes, std::map<std::string, StorePath> backMap)
+        : RefScanSink(std::move(hashes))
+        , backMap(backMap)
+    { }
```

Comment:
```suggestion
    PathRefScanSink(StringSet && hashes, std::map<std::string, StorePath> && backMap)
        : RefScanSink(std::move(hashes))
        , backMap(std::move(backMap))
    { }
```

Reply:
I would also move it out of the header into the implementation file with
```suggestion
    PathRefScanSink(StringSet && hashes, std::map<std::string, StorePath> backMap);
```
in just the header,

That leaks the least amount of what is basically implementation details given the "smart constructor".